encfs/BlockFileIO.cpp

```diff
@@ -191,7 +195,9 @@ bool BlockFileIO::write(const IORequest &req) {
   if (req.offset > fileSize) {
     // extend file first to fill hole with 0's..
     const bool forceWrite = false;
-    padFile(fileSize, req.offset, forceWrite);
+    int res = padFile(fileSize, req.offset, forceWrite);
+    if (res)
```

Comment:
so a positive value is a failure here?  Is a positive value from ::write considered a failure?  Seems like a change in logic here.